drivers/counter/counter_mchp_xec.c

```diff
@@ -0,0 +1,413 @@
+/* counter_xec.c - Microchip XEC counter driver */
+
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_LEVEL CONFIG_COUNTER_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(counter_mchp_xec);
+
+#include <counter.h>
+#include <soc.h>
+#include <errno.h>
+#include <logging/log.h>
+
+#include "counter_mchp_xec.h"
+
+static int counter_xec_start(struct device *dev)
+{
+	volatile struct counter_xec_registers *counter =
+						COUNTER_XEC_REG_BASE(dev);
+
+	if (counter->control_reg.enable) {
+		return -EALREADY;
+	}
+
+	counter->control_reg.enable = 1;
+
+	LOG_DBG("Counter started");
```

Comment:
since counter is multi-instance it would be good to include that in the log message. There is a built-in support for that in the logger (see https://github.com/zephyrproject-rtos/zephyr/blob/77db273f6f84b3d94d34b5e72070ec6bc31e41cf/drivers/counter/counter_nrfx_rtc.c#L355 as an example). It allows log runtime filtering on instance level (can be useful if one want to log counter instance and have second instance in the system which runs much more frequently).

Reply:
I'd rather print the dev pointer instead.

Reply:
that is another option. instance level logging might be more difficult to use but has filtering capability which might be useful when multiple counters are used in the system.